### ISSUE TITLE
[BUG] `SetQuota` with REDIS_PASSWORD in `envs` fails in Nomad

### ISSUE THREAD

**Issue by hashworks** [juicedata/juicefs-csi-driver#762]
**What happened**:
I tested our Redis Cluster and MinIO cluster setup locally:
```sh
REDIS_PASSWORD="redacted/redacted+redacted-redacted=redacted" juicefs format --storage minio --bucket 'https://minio.example.net/juicefs' --access-key redacted --secret-key redacted 'redis://10.0.0.141:6379,10.0.0.139:6379,10.0.0.142:6379,10.0.0.140:6379,10.0.0.145:6379,10.0.0.143:6379,10.0.0.146:6379,10.0.0.144:6379/0' jfs-test
# works
REDIS_PASSWORD="redacted/redacted+redacted-redacted=redacted" juicefs mount 'redis://10.0.0.141:6379,10.0.0.139:6379,10.0.0.142:6379,10.0.0.140:6379,10.0.0.145:6379,10.0.0.143:6379,10.0.0.146:6379,10.0.0.144:6379/0' /tmp/foo
# works
```

This worked fine, so I installed the controller and nodes in Nomad and created a volume with the following spec:
```hcl
type      = "csi"
plugin_id = "juicefs"

id   = "jfs-test"
name = "jfs-test"
capacity_max = "20G"
capacity_min = "10G"

capability {
  access_mode     = "multi-node-multi-writer"
  attachment_mode = "file-system"
}

secrets {
  name       = "jfs-test"
  metaurl    = "redis://10.0.0.141:6379,10.0.0.139:6379,10.0.0.142:6379,10.0.0.140:6379,10.0.0.145:6379,10.0.0.143:6379,10.0.0.146:6379,10.0.0.144:6379/0"
  bucket     = "https://minio.example.net/juicefs"
  storage    = "minio"
  access-key = "redacted"
  secret-key = "redacted"
  envs       = "{\"REDIS_PASSWORD\": \"redacted/redacted+redacted-redacted=redacted\"}"
}
```

Creating a job that used said volume failed with `NOAUTH`:
```
Setup Failure: failed to setup alloc: pre-run hook "csi_hook" failed: mounting volumes: node plugin returned an internal error, check the plugin allocation logs for more information: rpc error: code = Internal desc = set quota: 2023/09/28 17:32:20.520051 juicefs[55] <INFO>: Meta address: redis://10.0.0.141:6379,10.0.0.139:6379,10.0.0.142:6379,10.0.0.140:6379,10.0.0.145:6379,10.0.0.143:6379,10.0.0.146:6379,10.0.0.144:6379/0 [interface.go:497] redis: 2023/09/28 17:32:20 cluster.go:1760: getting command info: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time redis: 2023/09/28 17:32:20 cluster.go:1760: getting command info: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time 2023/09/28 17:32:20.531550 juicefs[55] <WARNING>: parse info: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time [redis.go:3547] redis: 2023/09/28 17:32:20 cluster.go:1760: getting command info: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time redis: 2023/09/28 17:32:20 cluster.go:1760: getting command info: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time 2023/09/28 17:32:20.537909 juicefs[55] <FATAL>: Load setting: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time [quota.go:127] : exit status 1
```

Weirdly, if I prefix the password with `:` (envs to `"{\"REDIS_PASSWORD\": \":redacted/redacted+redacted-redacted=redacted\"}"`, I get `WRONGPASS`:
```
Setup Failure: failed to setup alloc: pre-run hook "csi_hook" failed: mounting volumes: node plugin returned an internal error, check the plugin allocation logs for more information: rpc error: code = Internal desc = Could not mount juicefs: juicefs format error: 2023/09/28 17:36:37.271537 juicefs[19] <INFO>: Meta address: redis://10.0.0.141:6379,10.0.0.139:6379,10.0.0.142:6379,10.0.0.140:6379,10.0.0.145:6379,10.0.0.143:6379,10.0.0.146:6379,10.0.0.144:6379/0 [interface.go:497] redis: 2023/09/28 17:36:37 cluster.go:1760: getting command info: WRONGPASS invalid username-password pair or user is disabled. redis: 2023/09/28 17:36:37 cluster.go:1760: getting command info: WRONGPASS invalid username-password pair or user is disabled. 2023/09/28 17:36:37.280656 juicefs[19] <WARNING>: parse info: WRONGPASS invalid username-password pair or user is disabled. [redis.go:3547] redis: 2023/09/28 17:36:37 cluster.go:1760: getting command info: WRONGPASS invalid username-password pair or user is disabled. redis: 2023/09/28 17:36:37 cluster.go:1760: getting command info: WRONGPASS invalid username-password pair or user is disabled. 2023/09/28 17:36:37.288545 juicefs[19] <FATAL>: Load metadata: WRONGPASS invalid username-password pair or user is disabled. [format.go:453] : exit status 1
```

**What you expected to happen**:

Setting `envs` works just as with the `juicefs` command line.

**How to reproduce it (as minimally and precisely as possible)**:

Set up a redis cluster without ACLs, set up a MinIO cluster, set up a Nomad cluster. Create the CSI jobs and create the volume as shown above. Then create a job that uses it:
```hcl
job "test" {
  type = "service"

  group "test" {
    volume "jfs-test" {
      type            = "csi"
      source          = "jfs-test"
      attachment_mode = "file-system"
      access_mode     = "multi-node-multi-writer"
    }

    task "test" {
      driver = "exec"

      config {
        command = "sleep"
        args    = ["600"]
      }

      volume_mount {
        volume      = "jfs-test"
        destination = "/data/jfs-test"
      }
    }
  }
}
```

**Anything else we need to know?**

**Environment**:
- JuiceFS CSI Driver version: 0.22.1
- Nomad version: 1.6.2
- Object storage: MinIO
- Metadata engine info: Redis Server 7.0.5, cluster without ACLs, self-maintained

**Comment by hashworks** [comment]
I think the initial authentication works, but `set quota` isn't using the password?

**Comment by zxh326** [comment]
fixed in [v0.23.5](https://github.com/juicedata/juicefs-csi-driver/releases/tag/v0.23.5)
